src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -209,11 +212,20 @@ internal ItemManager(RecentItemsManager recentItemsManager)
                 return null;
             }
 
-            if (initialListOfItemsToBeIncluded.Count == 0)
+            if (builder.Count == 0)
             {
                 return HandleAllItemsFilteredOut(reason, data.SelectedFilters, completionRules);
             }
 
+            var initialListOfItemsToBeIncluded = builder.ToImmutableArray();
+
+            var experimentationService = document.Project.Solution.Workspace.Services.GetService<IExperimentationService>();
+            if (experimentationService.IsExperimentEnabled(WellKnownExperimentNames.SortCompletionListByMatch))
+            {
+                // Need a stable sorting algorithm to preserve the original order for items with same pattern match score.
+                initialListOfItemsToBeIncluded = initialListOfItemsToBeIncluded.OrderBy(new NullablePatternMatchComparer()).ToImmutableArray();
+            }
+
             var options = document?.Project.Solution.Options;
             var highlightMatchingPortions = options?.GetOption(CompletionOptions.HighlightMatchingPortionsOfCompletionListItems, document.Project.Language) ?? true;
```

Comment:
> var highlightMatchingPortions = options?.GetOption(CompletionOptions.HighlightMatchingPortionsOfCompletionListItems, document.Project.Language) ?? true; [](start = 11, length = 153)

You may want to call this before and then call PatternMatches above with this flag.

Reply:
Fixed